pkg/gatherers/common/gather_logs.go

```diff
@@ -17,25 +17,32 @@ import (
 	"github.com/openshift/insights-operator/pkg/utils/marshal"
 )
 
-// LogContainersFilter allows you to filter containers
-type LogContainersFilter struct {
-	Namespace                string
-	LabelSelector            string
-	FieldSelector            string
-	ContainerNameRegexFilter string
-	MaxNamespaceContainers   int
+// LogResourceFilter allows you to filter containers
+type LogResourceFilter struct {
+	Namespace                string `json:"namespace"`
+	LabelSelector            string `json:"label_selector,omitempty"`
+	FieldSelector            string `json:"field_selector,omitempty"`
+	ContainerNameRegexFilter string `json:"container_name_regex_filter,omitempty"`
+	PodNameRegexFilter       string `json:"pod_name_regex_filter,omitempty"`
+	MaxNamespaceContainers   int    `json:"max_namespace_containers,omitempty"`
 }
 
 // LogMessagesFilter allows you to filter messages
 type LogMessagesFilter struct {
-	MessagesToSearch []string
-	IsRegexSearch    bool
-	SinceSeconds     int64
-	LimitBytes       int64
-	TailLines        int64
-	Previous         bool
+	MessagesToSearch []string `json:"messages_to_search,omitempty"`
+	IsRegexSearch    bool     `json:"is_regex_search,omitempty"`
+	SinceSeconds     int64    `json:"since_seconds,omitempty"`
+	LimitBytes       int64    `json:"limit_bytes,omitempty"`
+	TailLines        int64    `json:"tail_lines,omitempty"`
+	Previous         bool     `json:"previous,omitempty"`
 }
 
+// FilterLogCallbackFunc is used to apply extra filtering to log lines
+type FilterLogCallbackFunc func(lines []string) []string
+
+// FilterLogOption is used to apply the filter engine (e.g: Regex, Substring)
+type FilterLogOption func(line string) bool
```

Comment:
It's not clear to me what the return value means? Filter or not filter?

Reply:
// FilterLogOption is a functional option used to apply the filtering engine, such as Regex or Substring matching,
// to determine whether a given log line satisfies the specified condition.
// When applied, it returns true if the provided regex or substring is found in the log line, indicating a match.
// Conversely, it returns false if no match is detected.